go/pkg/client/client.go

```diff
@@ -522,6 +522,20 @@ func (c *Client) GetCapabilities(ctx context.Context, req *repb.GetCapabilitiesR
 	if err != nil {
 		return nil, err
 	}
+	// If the CAS server is different to the execution server, we need to get its capabilities separately.
```

Comment:
Since this function is not yet used anywhere, we still have freedom to decide what would be best. I think the behavior is a bit unintuitive, so it's worth splitting out explicitly:
- one wrapper function, GetBackendCapabilities that accepts a connection as parameter (it's not really that important to persist the regrpc.CapabilitiesClient as instance variable, this will only be called once-per-client anyway)
- another wrapper function, GetCapabilities, that only accepts the context and the instance_name and returns the merged ServerCapabilities.
WDYT?

Reply:
Sorry for the slow response - been a busy week here.
That makes sense to me, I think it would be clearer for a user what's going on. Will update this PR shortly.